packages/dates/src/apply-time-zone-offset.ts

```diff
@@ -5,10 +5,14 @@ export function applyTimeZoneOffset(
   timeZone1?: string,
   timeZone2?: string,
 ) {
+  const dateToCheckForDSTOffset = new Date(date);
+  dateToCheckForDSTOffset.setHours(2);
+
+  const dstOffsetDiff = date.getTimezoneOffset() - dateToCheckForDSTOffset.getTimezoneOffset();
   const initialOffset = getTimeZoneOffset(date, timeZone1, timeZone2);
   const adjustedDate = new Date(date.valueOf() - initialOffset * 60 * 1000);
   const targetOffset = getTimeZoneOffset(adjustedDate, timeZone1, timeZone2);
-  const offsetDiff = targetOffset - initialOffset;
+  const offsetDiff = targetOffset - initialOffset - Math.abs(dstOffsetDiff);
```

Comment:
Why are we always subtracting the absolute value of the offset?

Reply:
I couldn't really think of a better way of doing this, but in order to have it properly calculate the jump forward and back, we need to subtract the absolute value of the difference otherwise the calculation will be wrong on the jump forward.

Reply:
This feels weird. An offset is an offset. You should be able to just add (or subtract it) based on how the offset is calculated. Could you maybe illustrate that problem by having two unit tests, one for the positive offset case and one for the negative offset case?

Reply:
You know what, I totally goofed. It can be written like so: `const offsetDiff = targetOffset - initialOffset  + dstOffsetDiff;`

Reply:
Ok sorry for stringing this along. I have refactored it to check if it needs a DST offset of `60`. DST will only ever be an offset of an hour.

Reply:
> DST will only ever be an offset of an hour.

Nope.  😢 

Granted, such edge cases are rare, but they [do](https://www.zainrizvi.io/blog/falsehoods-programmers-believe-about-time-zones/) [happen](https://en.wikipedia.org/wiki/Time_in_Russia).